doc/source/admin/data.md

```diff
@@ -0,0 +1,595 @@
+can be placed ``file_source_templates.yml`` in Galaxy configuration directory (or any path
+pointed to by the configuration option ``file_source_templates_config_file`` in ``galaxy.yml``).
+Alternatively, the  configuration can be placed directly into ``galaxy.yml`` using the
+``file_source_templates`` configuration option.
+
+
+### File Source Types
+
+#### ``posix``
+
+The syntax for the ``configuration`` section of ``posix`` templates looks like this.
+
+![](file_source_posix_configuration_template.png)
+
+At runtime, after the ``configuration`` template is expanded, the resulting dictionary
+passed to Galaxy's file source plugin infrastructure looks like this and should match a subset
+of what you'd be able to add directly to ``file_sources_conf.yml`` (Galaxy's global file source
+configuration).
+
+![](file_source_posix_configuration.png)
+
+#### ``s3fs``
+
+```{literalinclude} ../../../lib/galaxy/files/templates/examples/production_s3fs.yml
+:language: yaml
+```
+
+```{literalinclude} ../../../lib/galaxy/files/templates/examples/production_aws_public_bucket.yml
+:language: yaml
+```
+
+```{literalinclude} ../../../lib/galaxy/files/templates/examples/production_aws_private_bucket.yml
+:language: yaml
+```
+
+![](file_source_s3fs_configuration_template.png)
+
+At runtime, after the ``configuration`` template is expanded, the resulting dictionary
+passed to Galaxy's file source plugin infrastructure looks like this and should match a subset
+of what you'd be able to add directly to ``file_sources_conf.yml`` (Galaxy's global file source
+configuration).
+
+![](file_source_s3fs_configuration.png)
+
+#### ``ftp``
+
+```{literalinclude} ../../../lib/galaxy/files/templates/examples/production_ftp.yml
+:language: yaml
+```
+
+The syntax for the ``configuration`` section of ``ftp`` templates looks like this.
+
+![](file_source_ftp_configuration_template.png)
+
+At runtime, after the ``configuration`` template is expanded, the resulting dictionary
+passed to Galaxy's file source plugin infrastructure looks like this and should match a subset
+of what you'd be able to add directly to ``file_sources_conf.yml`` (Galaxy's global file source
+configuration).
+
+![](file_source_ftp_configuration.png)
+
+#### ``azure``
+
+The syntax for the ``configuration`` section of ``azure`` templates looks like this.
+
+![](file_source_azure_configuration_template.png)
+
+At runtime, after the ``configuration`` template is expanded, the resulting dictionary
+passed to Galaxy's file source plugin infrastructure looks like this and should match a subset
+of what you'd be able to add directly to ``file_sources_conf.yml`` (Galaxy's global file source
+configuration).
+
+![](file_source_azure_configuration.png)
+
+### YAML Syntax
+
+![galaxy.files.templates.models](file_source_templates.png)
+
+### Ready To Use Production File Source Templates
+
+The templates have been tested by a Galaxy developer and are sufficiently generic that
```

Comment:
```suggestion
The templates are sufficiently generic that
```

Reply:
Yeah? 😆 . I'll make the change but it seems odd.